internal/controller/imagerepository_controller.go

```diff
@@ -181,6 +185,25 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	// Provision image repository if it hasn't been done yet
 	if !controllerutil.ContainsFinalizer(imageRepository, ImageRepositoryFinalizer) {
 		setMetricsTime(repositoryIdForMetrics, reconcileStartTime)
+		if isComponentLinked(imageRepository) {
+			componentExists, requeueAfterSeconds, err := r.CheckComponentExistence(ctx, imageRepository)
+			// getting component failed
```

Comment:
nitpick: the comments describe the situation if the condition is true. Does it make sense to have the comment inside the if block?

Reply:
ok